enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,343 @@
+	// +optional
+	MaxDeviceCount *int32 `json:"maxDeviceCount"`
+	// VolumeMode determines whether the PV created is Block or Filesystem. By default it will
+	// be block
+	// + optional
+	VolumeMode PersistentVolumeMode `json:"volumeMode,omitempty"`
+	// FSType type to create when volumeMode is Filesystem
+	// +optional
+	FSType string `json:"fsType,omitempty"`
+	// If specified, a list of tolerations to pass to the discovery daemons.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// DeviceInclusionSpec is the filtration rule for including a device in the device discovery
+	// +optional
+	DeviceInclusionSpec *DeviceInclusionSpec `json:"deviceInclusionSpec"`
+}
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational or nonRotational
+type DeviceMechanicalProperty string
+
+const (
+	// The mechanical properties of the devices
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "Rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+type DeviceType string
+
+const (
+	// The DeviceTypes that will be supported by the LSO.
+	// These Discovery policies will based on lsblk's type output
+	// Disk represents a device-type of disk
+	Disk DeviceType = "Disk"
+	// Part represents a device-type of partion
+	Partition DeviceType = "Partition"
+)
+
+type DeviceInclusionSpec struct {
+	// Devices is the list of devices that should be used for automatic detection.
+	// This would be one of the types supported by the local-storage operator. Currently,
+	// the supported types are: disk, part. If the list is empty no devices will be selected.
+	Devices []DeviceType `json:"devices"`
+
+	// DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+	// by default, it selects both
+	// +optional
+	DeviceMechanicalProperties []DeviceMechanicalProperty `json:"deviceMechanicalProperties"`
+
+	// MinSize is the minimum size of the device which needs to be included
+	// +optional
+	MinSize *resource.Quantity `json:"minSize"`
+
+	// MaxSize is the maximum size of the device which needs to be included
+	// +optional
+	MaxSize *resource.Quantity `json:"maxSize"`
+
+	// Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Models []string `json:"models"`
+
+	// Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Vendors []string `json:"vendors"`
+}
+
+type LocalVolumeGroupCondition string
+
+const (
+	ActivePhase LocalVolumeGroupCondition = "Active"
+	FailedPhase LocalVolumeGroupCondition = "Failed"
+)
+
+type LocalVolumeGroupStatus struct {
+
+	// Conditions is the state of the LocalVolumeGroup
+	Condition LocalVolumeGroupCondition `json:"conditions,omitempty"`
```

Comment:
why did we rename this variable from phase to condition? In k8s typically conditions have special meaning.

Reply:
you had `[]operatorv1.Condition` and it was correct, can you please restore it?

Reply:
And remove `Message` and `Reason` below, they're part of `Condition` already.

Reply:
changed back to phase. Let's add condition in a future release.

Reply:
yeah updated